Make it work!

osfanbuff63 opened the pull request:
Too many changes to count, I've been working on this for the last couple of days xD

Contributions welcome, as always!

### Bugs to fix

- [x] Backslashes (`\`) rendering as ¥ for some reason
- [x] Always errors even on success (with error `None`)
- [x] Try to fix dark mode

### Features to add

- [ ] Make an FO-specific Minecraft instance (WIP)
- [ ] (auto) Updater

osfanbuff63 committed:
:sparkles: It finally works!

Too many changes to count, I've been working on this for the last couple of days xD

Kichura commented on .github/workflows/pylint.yml:
```diff
@@ -7,17 +7,29 @@ jobs:
+          "3.8",
+          "3.9",
+          "3.10",
+          #"3.11" soon™
```
`# "3.11" soon™`

This would've looked a bit fancier imo.

Kichura commented on README.md:
```diff
@@ -1,53 +1,62 @@
 
 - Pros
-    - Can always used with the newest commits
+  - Can always used with the newest commits
```
`Can always be used with the newest commits`

Since this part has a typo.

osfanbuff63 replied:
Can't believe I didn't catch that lol

Kichura commented on data/theme.txt:
```diff
@@ -1 +1 @@
-dark
\ No newline at end of file
+light
```
You became the thing you swore to destroy! (joke btw)

osfanbuff63 replied:
There's some sort of dark mode problem, haven't put the time into it yet

This is supposed to be gitignored too :O

Kichura commented on docs/support.md:
````diff
@@ -1,50 +1,61 @@
+Don't worry! Try the following commands (AFTER trying [install/linux.sh](install/linux.sh)):
 
-```
+```bash
````
Shouldn't this also have `#!/bin/bash` or `#!/usr/bin/env/bash` at beginning of every bash script?

osfanbuff63 replied:
Probably yeah

Kichura reviewed (commented):
A few questions to be made for this PR as it could be explained a little from OP.

osfanbuff63 committed:
:memo: Clear up some things

Kichura reviewed (approved):
looks good so far.

osfanbuff63 committed:
:see_no_evil: No more __pycache__

osfanbuff63 committed:
:construction_worker: Don't run git repo sync on anything but main

RozeFound commented on vanilla-installer/main.py:
```diff
@@ -1,35 +1,37 @@
 
+PATH_FILE = str(pathlib.Path("data/mc-path.txt").resolve())
+FOLDER_LOC = ""
+TEMP_FOLDER = str(pathlib.Path("/temp").resolve())
```
It will likely transform to `temp` folder in `root` directory, are you aware?

osfanbuff63 replied:
Actually, that was probably meant to be a `temp/`.

Ideally there's an OS method for this somewhere, but I'll look into that.

RozeFound replied:
[This](https://docs.python.org/3/library/tempfile.html) maybe?

osfanbuff63 replied:
Turned out that this is only used by `download_fabric()`, so this was easy to implement. Before this PR it was also used for `download_pack`, but that can't be used now because of Packwiz installer limitations.

Ended up reverting the change since that apparently deletes the directory after the code in the with statement is finished and running the installer is for some reason run by a different function (I guess since it takes different parameters?) so just made a temp directory with `pathlib`.

RozeFound commented on vanilla-installer/main.py:
```diff
@@ -1,35 +1,37 @@
-        open(PATH_FILE, "w").write(path)
+        open(PATH_FILE, "w", encoding="utf-8").write(path)
         return path
+    return
```
Return type is `str`, but you return `None`?

osfanbuff63 replied:
Should probably redo this here. Ideally this makes it go back to the previous path if None is returned, but we should probably just rerun mll.utils.get_minecraft_path here so that the path doesn't appear as none.

RozeFound replied:
Or you can do `() -> str | None` to keep things clear

osfanbuff63 replied:
Alright, done.

RozeFound commented on vanilla-installer/main.py:
```diff
@@ -73,26 +98,41 @@ def text_update(text: str, widget=None, mode: str = "info") -> None:
+    Returns:
+        str: The output of the command.
+    """
+    output = str(logging.debug(subprocess.check_output(text.split()).decode("utf-8")))
```
Isn't `.decode()` returns `str` by itself?

osfanbuff63 replied:
This really wasn't my code, so ¯\\_(ツ)_/¯

I probably just reformatted it or something.

RozeFound replied:
Your emoticon lost his hand... Anyway it's way too nested in the first place, you should probably split it to a few more lines

osfanbuff63 replied:
Alright, should be fixed as of latest commit.

RozeFound replied:
Now he lost both of his shoulders... Why do you need specifically `logging.debug` output?

RozeFound commented on vanilla-installer/main.py:
```diff
@@ -120,64 +167,99 @@ def install_fabric(
-        "https://api.github.com/repos/Fabulously-Optimized/fabulously-optimized/releases/latest"
-    ).json()
-    pack_file = ""
+    Downloads both the packwiz_install_bootstrap and the packwiz_installer jars.
```
Why? bootstrap will download and launch installer by itself.

osfanbuff63 replied:
Didn't know that actually, I'll just download bootstrap then

RozeFound replied:
Well that's what bootstrap is for, to download and keep updated packwiz-installer

osfanbuff63 replied:
Fixed as of latest commit

RozeFound commented:
And about backslashes, you can just use forward ones, it works even in windows and looks cleaner in my opinion.

nsde commented:
Happy to see this project isn't dead! When I'm back from my vacation, I'll take a look at it and maybe work further as I haven't played Minecraft in a while and this is the perfect oppurtunity.

Thanks to everyone contributing!

osfanbuff63 commented on vanilla-installer/main.py:
```diff
@@ -73,26 +98,41 @@ def text_update(text: str, widget=None, mode: str = "info") -> None:
     return output
 
 
+# This function isn't ever called, it's probably better to depreciate it (feedback wanted)
```
Anyone have thoughts here?

RozeFound replied:
Well, seems pretty useless, at least in current state... 
You should rewrite it to something more useful than just opening web page, or remove it completely since it's not used anywhere anyway

osfanbuff63 replied:
Removed.

osfanbuff63 committed:
:bug: Fix some things and add some suggestions

osfanbuff63 committed:
:egg: Revert some of the last commit

Turns out tempfile.TemporaryDirectory deletes the directory once the code in the with statement is done running. Also adds an easter egg :P

RozeFound commented on vanilla-installer/main.py:
```diff
@@ -133,9 +134,11 @@ def download_fabric(
-    installers = requests.get("https://meta.fabricmc.net/v2/versions/installer").json()
-    download = requests.get(installers[0]["url"])
-    file_path = TEMP_FOLDER + download.url.split("/")[-1]
+    with tempfile.TemporaryDirectory as tmpdir:
```
You forgot `()`

RozeFound replied:
Oh, you already removed it, okay

osfanbuff63 replied:
yeah see the other comments

RozeFound replied:
Than that's what you need prob https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp

osfanbuff63 replied:
Applied in latest commit.

osfanbuff63 committed:
More suggestions

RaptaG commented:
I'll have to adapt this I assume..
Eh, I'll wait for it to be merged and then I will.

osfanbuff63 committed:
Begin major changes (pipenv, our own instance)

osfanbuff63 committed:
Fix Python 3.11 failing

osfanbuff63 committed:
Fix lint error

osfanbuff63 committed:
Remove symlinks

osfanbuff63 committed:
:bug: Fix dark mode

it was such a simple fix lol

osfanbuff63 committed:
Disable `win.maxsize` and docs improvements

osfanbuff63 committed:
Remove references to symlinks

osfanbuff63 committed:
More bug fixing

osfanbuff63 committed:
why did this not commit

osfanbuff63 committed:
Attempt more fixes and improve docs

osfanbuff63 committed:
Fix actions and improve code

osfanbuff63 committed:
:ioa:

osfanbuff63 committed:
why actions

osfanbuff63 committed:
Bump pylint down and fix actions again

osfanbuff63 committed:
pipenv run (again)

osfanbuff63 committed:
Fix again

osfanbuff63 committed:
.....so tired of this

osfanbuff63 committed:
I give up lol

osfanbuff63 committed:
More docs

nsde reviewed (approved):
:+1:
